drivers/isbmex.c

```diff
@@ -175,7 +175,7 @@ static const char *getpacket(int *we_know){
   }
 
   r=read(upsfd,buf,255);
-  D(printf("%d bytes read: ",r);)
+  D(printf("%ld bytes read: ",r);)
```

Comment:
There is no big guarantee about `(s)size_t` and `(long) int` byte sizes (I believe it is "just some integer type"), so it should be printed with `%zd` (`ssize_t`) or `%zu` (`size_t`) from C99 onwards, or cast into an `intmax_t` and `%jd`.

See e.g. https://stackoverflow.com/a/5251918 for details.

- [ ] Fix fixed printing of ssize_t

Reply:
What is the minimum C dialect required?  i.e. can >= C99 be assumed?  I don't find any other "%z" in the codebase currently.

Reply:
Good question. I did not find a direct statement on that. In code I see a couple of commits from 2008-2010 allowing to build on compilers with standards before C99, of those fixes at least one is still in codebase:
https://github.com/networkupstools/nut/blob/master/configure.ac#L233

But then from 2011 up to some of my 2017 commits, there were a few fixes to get this working specifically with C99+ and at least for mine nobody complained or "broke back" AFAIK.

Reply:
C99 is the last REQ we expressed
not sure where we said so, probably a volatile mail on the ML, plus the configure check AC_PROG_CC_C99.
That relates to old Unix systems, that we still (?) support. A refreshed audit of the situation here would be good

Reply:
I played on my workstation with Debian Linux (started life as v8, but updated since so mixed-version) and tried different `-std=...` flags to GCC, like this:

````
:; make distclean ; PATH=/usr/lib/ccache:$PATH CC=/usr/lib/ccache/gcc CXX=/usr/lib/ccache/g++ \
    ./configure --with-cgi CFLAGS="-std=gnu99" CXXFLAGS="-std=gnu++99" && \
    (make -j8 all -k 2>/dev/null 1>&2) ; echo ======================================== ; make all -k
````

It failed with all strictly-C flags (`-ansi` aka `-stc=c89`, c90, c99, c11) mostly due to Linux system headers not really working with that, missing some basic routine definitions or having confusing parameter expectations as a result.

It fared better for GNU C dialects, though in pedantic mode exposed some issues like those you fixed and more (printing into buffer too small, variadic macros requiring at least one argument, unused vars, possibly botched logical conditions where parentheses may be due...) with all dialects, so we should fix those eventually. Beside that...

For `-std=gnu90` it reported a few offenses seemingly easy to fix:
* declarations inside `for` like `for (int i=0; ...)` in solis.c
* BUT quite a few warnings so binary code might not work right in the end:
** format string mismatches that your PR probably fixes
** format chars introduced in later standard versions
** initializer element is not computable at load time
** ISO C90 forbids mixed declarations and code
** undefined `__func__` interpreted as `int not %s` for logging => probably would segfault at runtime
** C++ style comments with `//` in a few spots
** (pedantic) comma at end of enumerator list

...so while restoring compatibility with C90 (GNU dialect at least) may be possible, it is a considerable effort and not sure it is worth doing unless someone actually suffers from the lack. So far nobody complained AFAIK.

For `-std=gnu++11` there is a deprecation warning for
````
In file included from nutclient.cpp:20:0:
nutclient.h:154:79: error: dynamic exception specifications are deprecated in C++11 [-Werror=deprecated]
  virtual void authenticate(const std::string& user, const std::string& passwd)throw(NutException)=0;
````


````
:; make distclean ; PATH=/usr/lib/ccache:$PATH CC=/usr/lib/ccache/gcc CXX=/usr/lib/ccache/g++ \
    ./configure --with-cgi CFLAGS="-Wall -Werror -pedantic -std=gnu99" CXXFLAGS="-Wall -Werror -std=gnu++99" && \
    (make -j8 all -k 2>/dev/null 1>&2) ; echo ======================================== ; make all -k
````

Reply:
OK, since >= C99 is required I'll switch to %z.

Reply:
With #823 / #824 the current NUT master should test a dozen more combos (relatively arbitrarily chosen) on Travis to aid in code cleanup efforts like this. Thanks @seanm for triggering this ;)